src/mina-signer/src/sign-zkapp-command.unit-test.ts

```diff
@@ -0,0 +1,297 @@
+    let zkappCommandJson = ZkappCommand.toJSON(zkappCommand);
+
+    // snarkyjs fromJSON -> toJSON roundtrip, + consistency with mina-signer
+    let zkappCommandSnarky = ZkappCommandSnarky.fromJSON(zkappCommandJson);
+    let zkappCommandJsonSnarky = ZkappCommandSnarky.toJSON(zkappCommandSnarky);
+    expect(JSON.stringify(zkappCommandJson)).toEqual(
+      JSON.stringify(zkappCommandJsonSnarky)
+    );
+    let recoveredZkappCommand = ZkappCommand.fromJSON(zkappCommandJson);
+    fixEmptyPermissions(recoveredZkappCommand);
+    expect(recoveredZkappCommand).toEqual(zkappCommand);
+
+    // tx commitment
+    let ocamlCommitments = Ledger.transactionCommitments(
+      JSON.stringify(zkappCommandJson)
+    );
+    let callForest = accountUpdatesToCallForest(zkappCommand.accountUpdates);
+    let commitment = callForestHash(callForest);
+    expect(commitment).toEqual(ocamlCommitments.commitment.toBigInt());
+
+    let memo = Memo.fromBase58(memoBase58);
+    let memoHash = Memo.hash(memo);
+    let memoHashSnarky = Ledger.memoHashBase58(memoBase58);
+    expect(memoHash).toEqual(memoHashSnarky.toBigInt());
+
+    let feePayerAccountUpdate = accountUpdateFromFeePayer(feePayer);
+    let feePayerJson = AccountUpdate.toJSON(feePayerAccountUpdate);
+
+    let feePayerInput = AccountUpdate.toInput(feePayerAccountUpdate);
+    let feePayerInput1 = inputFromOcaml(
+      toJSON(Ledger.hashInputFromJson.body(JSON.stringify(feePayerJson.body)))
+    );
+    expect(stringify(feePayerInput.fields)).toEqual(
+      stringify(feePayerInput1.fields)
+    );
+    expect(stringify(feePayerInput.packed)).toEqual(
+      stringify(feePayerInput1.packed)
+    );
+
+    let feePayerDigest = feePayerHash(feePayer);
+    expect(feePayerDigest).toEqual(ocamlCommitments.feePayerHash.toBigInt());
+
+    let fullCommitment = hashWithPrefix(prefixes.accountUpdateCons, [
+      memoHash,
+      feePayerDigest,
+      commitment,
+    ]);
+    expect(fullCommitment).toEqual(ocamlCommitments.fullCommitment.toBigInt());
+
+    // signature
+    let sigTestnet = signFieldElement(fullCommitment, feePayerKey, 'testnet');
+    let sigMainnet = signFieldElement(fullCommitment, feePayerKey, 'mainnet');
+    let sigTestnetOcaml = Ledger.signFieldElement(
+      ocamlCommitments.fullCommitment,
+      feePayerKeySnarky,
+      false
+    );
+    let sigMainnetOcaml = Ledger.signFieldElement(
+      ocamlCommitments.fullCommitment,
+      feePayerKeySnarky,
+      true
+    );
+    expect(Signature.toBase58(sigTestnet)).toEqual(sigTestnetOcaml);
+    expect(Signature.toBase58(sigMainnet)).toEqual(sigMainnetOcaml);
+
+    let verify = (s: Signature, id: NetworkId) =>
+      verifyFieldElement(s, fullCommitment, feePayerAddress, id);
+    expect(verify(sigTestnet, 'testnet')).toEqual(true);
+    expect(verify(sigTestnet, 'mainnet')).toEqual(false);
+    expect(verify(sigMainnet, 'testnet')).toEqual(false);
+    expect(verify(sigMainnet, 'mainnet')).toEqual(true);
```

Comment:
I already checked that the signatures are equal to the ones created from OCaml. So OCaml verification will yield the same unless the OCaml implementation has a bug. I'd like to leave that to the OCaml-side unit tests because OCaml verification is not, at the moment, exposed to snarkyjs